ansible/tasks/setup-coredumps.yml

```
@@ -0,0 +1,230 @@
+      new="${new} coredump_filter=49"
+      sed -i "s/^GRUB_CMDLINE_LINUX_DEFAULT=.*/GRUB_CMDLINE_LINUX_DEFAULT=\"${new}\"/" "$file"
+    done
+  when: 
+    - stage2_nix
+    - grub_d_files.stdout != ""
+  register: grub_d_updated
+
+- name: Update GRUB after configuration changes
+  become: yes
+  command: update-grub
+  when: 
+    - stage2_nix
+    - (grub_main_updated is defined and grub_main_updated.changed) or (grub_d_updated is defined and grub_d_updated.changed)
+
+- name: Install gdb for debugging
+  become: yes
+  apt:
+    pkg:
+      - gdb
+    state: present
+    update_cache: yes
+    cache_valid_time: 3600
+  when: stage2_nix
+
+- name: Create root .gdbinit configuration
+  become: yes
+  copy:
+    content: |
+      # GDB configuration for PostgreSQL debugging
+      # Note: debug-file-directory and substitute-path need to be set manually
+      # after installing debug symbols and source files from nix
+      set auto-load safe-path /
+      # Example paths - will be set dynamically when debugging:
+      # set debug-file-directory /nix/store/<hash>-postgresql-<version>-debug/lib/debug/
+      # set substitute-path ./ /nix/store/<hash>-postgresql-<version>-src-<version>/
+    dest: /root/.gdbinit
+    mode: '0644'
+    owner: root
+    group: root
+  when: stage2_nix
+
+- name: Create postgres user .gdbinit configuration
+  become: yes
+  become_user: postgres
+  copy:
+    content: |
+      # GDB configuration for PostgreSQL debugging
+      # Note: debug-file-directory and substitute-path need to be set manually
+      # after installing debug symbols and source files from nix
+      set auto-load safe-path /
+      # Example paths - will be set dynamically when debugging:
+      # set debug-file-directory /nix/store/<hash>-postgresql-<version>-debug/lib/debug/
+      # set substitute-path ./ /nix/store/<hash>-postgresql-<version>-src-<version>/
+    dest: /var/lib/postgresql/.gdbinit
+    mode: '0644'
+    owner: postgres
+    group: postgres
+  when: stage2_nix
+
+- name: Create coredump setup documentation
+  become: yes
+  copy:
+    content: |
+      # PostgreSQL Coredump Configuration
+      
+      This AMI has been configured to collect PostgreSQL coredumps for debugging.
+      
+      ## Configuration Files Modified:
+      - /etc/sysctl.d/99-coredump.conf - Kernel coredump settings
+      - /etc/systemd/system.conf.d/50-coredump.conf - Systemd coredump settings
+      - /etc/security/limits.conf - User limits for coredump size
+      - /etc/default/grub - GRUB configuration for coredump_filter
+      
+      ## Coredump Location:
+      Coredumps are saved to: /pg_coredump_debug/
+      
+      ## Debug Symbols:
+      Debug symbols and source files are already installed via nix during the build.
+      To find the paths:
+      ```
+      sudo -u postgres nix profile list | grep postgresql_.*_debug
+      sudo -u postgres nix profile list | grep postgresql_.*_src
```

I had to add `-I`, otherwise I get error `sudo: nix: command not found`:
```suggestion
      sudo -i -u postgres nix profile list | grep postgresql_.*_debug
      sudo -i -u postgres nix profile list | grep postgresql_.*_src
```